src/Joint.cc

```diff
@@ -166,18 +166,41 @@ Errors Joint::Load(ElementPtr _sdf)
         "] was specified for both."});
   }
 
-  if (_sdf->HasElement("axis"))
+  // Map of follower axis name to dataPtr->axis[] array index
+  // The keys of this map are also the only valid leader axis names
+  const std::map<std::string, std::size_t>
+      followerAxisNames = {{"axis", 0}, {"axis2", 1}};
+  for (const auto &followerAxisNameToIndex : followerAxisNames)
   {
-    this->dataPtr->axis[0].emplace();
-    Errors axisErrors = this->dataPtr->axis[0]->Load(_sdf->GetElement("axis"));
-    errors.insert(errors.end(), axisErrors.begin(), axisErrors.end());
-  }
-
-  if (_sdf->HasElement("axis2"))
-  {
-    this->dataPtr->axis[1].emplace();
-    Errors axisErrors = this->dataPtr->axis[1]->Load(_sdf->GetElement("axis2"));
-    errors.insert(errors.end(), axisErrors.begin(), axisErrors.end());
+    const std::string followerAxis = followerAxisNameToIndex.first;
+    const std::size_t i = followerAxisNameToIndex.second;
+    if (_sdf->HasElement(followerAxis))
+    {
+      this->dataPtr->axis[i].emplace();
+      Errors axisErrors =
+          this->dataPtr->axis[i]->Load(_sdf->GetElement(followerAxis));
```

Comment:
nit: We can avoid repetitions of `this->dataPtr->axis[i]` by
```suggestion
      auto &axis = this->dataPtr->axis[i].emplace();
      Errors axisErrors =
          axis.Load(_sdf->GetElement(followerAxis));
```

Reply:
good call: https://github.com/gazebosim/sdformat/pull/1166/commits/7fd569e420820d0668af2b7e3f342d3f46a9f5e2